features/netsocket/nsapi_dns.cpp

```diff
@@ -29,8 +33,51 @@
 #define DNS_BUFFER_SIZE 512
 #define DNS_TIMEOUT 5000
 #define DNS_SERVERS_SIZE 5
+#define DNS_RESPONSE_MIN_SIZE 12
+#define DNS_MAX_TTL 604800
+#define DNS_CACHE_SIZE 3
+#define DNS_STACK_SERVERS_NUM 5
+#define DNS_QUERY_QUEUE_SIZE 5
+
+struct DNS_CACHE {
+    SocketAddress address;
+    char host[128];
```

Comment:
If RFC 1035 states that full DNS name can be 255 bytes, could this non-standard limitation be specified by some configuration option. I have no objection for limiting this by default, but if the max length was exposed via a public define, the client application could use it for its own buffers too. And a space limited application might try to tune the max name length and cache size.

Reply:
I'll add configuration options to json for both name length and number of entries in cache.

Reply:
We have hit problems with very long Amazon server names - not sure what limit was being breached there. Maybe 64 characters in the ESP8266?

I'd like the default to be the standard if it possibly fits. For a cache with your own internal structures, maybe claw some back by using the nsapi_addr_t, skipping the string literal representation?

Reply:
Changed limit to 255 and changed storage to use dynamic memory for host name.